src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -76,7 +94,7 @@ public function supportsNormalization($data, $format = null)
      */
     public function denormalize($data, $class, $format = null, array $context = array())
     {
-        $dateTimeFormat = isset($context[self::FORMAT_KEY]) ? $context[self::FORMAT_KEY] : null;
+        $dateTimeFormat = $context[static::FORMAT_KEY] ?? null;
```

Comment:
Even before this PR, `$this->format` was ignored when denormalizing (but not when normalizing).
Changing this behavior introduce a BC break, and breaks the tests.

An alternative could be to introduce a new key, like `denormalize_datetime_format` and deprecate using `denormalize_datetime_format` when denormalizing.

Anyway, it's out of scope of this PR.